pkg/3scale/amp/operator/system_options_provider.go

```diff
@@ -378,11 +378,8 @@ func (s *SystemOptionsProvider) setBackendOptions() error {
 		return fmt.Errorf("'%s' field of '%s' secret must have 'scheme://user:password@host/path' format", component.BackendSecretBackendListenerServiceEndpointFieldName, component.BackendSecretBackendListenerSecretName)
 	}
 
-	if urlObj.Path == "" {
-		urlObj.Path = "/internal/"
-	}
-
-	s.options.BackendServiceEndpoint = urlObj.String()
+	urlObj.Path += "/internal/"
```

Comment:
This is not checking for possible trailing slashes in the base URL, so may potentially result in double slashes (which is not very critical, IMO).

In Go 1.19 the following can be done instead:
```
s.options.CoreInternalAPIEndpoint = urlObj.JoinPath("/internal/").String()
```

Reply:
We are not yet there (Go 1.19). We recently updated to 1.18 https://github.com/3scale/3scale-operator/pull/814

If we needed to upgrade the secret content (which I think it is not needed), I would remove this logic in the 3scale operator of adding `internal` and rely on the value of the secret. I do not know the reason why the operator needs to add this `/internal` to the path. 

However, we  do not need to upgrade the content of the secret, so I would just leave as it is:

```go
if urlObj.Path == "" {
  urlObj.Path = "/internal/"
}
s.options.CoreInternalAPIEndpoint = urlObj.String()
```

For me the initial intention was to allow any custom path if required. Only if the custom path is not there, the operator adds the `/internal`. With `urlObj.Path += "/internal/"` you are breaking that behavior.

Reply:
Please see my comment [here](https://github.com/3scale/3scale-operator/pull/821#issuecomment-1542065989) that provides more contexts.

With regards to how to append the `/internal/` path (only applicable if we go with the option 2), I think the option I suggested is more flexible. It permits the service URL for backend listener to have some path already.

So, if we set `https://my-apisonator-stub.somedomain.com/whatever` as `service_endpoint` (assuming apisonator is listening at that address), then the core API endpoint will be set to `https://my-apisonator-stub.somedomain.com/whatever/internal` correctly.

While, in the option you suggest it will not work (the `/internal` will not be appended).